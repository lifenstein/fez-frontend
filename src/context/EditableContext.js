import React from 'react';

export default React.createContext({
    editable: false,
});
