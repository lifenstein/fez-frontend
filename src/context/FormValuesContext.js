import React from 'react';
import Immutable from 'immutable';

const FormValuesContext = React.createContext({
    formValues: Immutable.Map({}),
    onDeleteAttachedFile: () => {},
    onRenameAttachedFile: () => {},
});

export default FormValuesContext;
