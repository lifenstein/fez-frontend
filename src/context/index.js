import { useContext } from 'react';

import AccountContext from './AccountContext';
import FormErrorsContext from './FormErrorsContext';
import FormValuesContext from './FormValuesContext';
import LocallyStoredReducerContext from './LocallyStoredReducerContext';
import OrcidSyncContext from './OrcidSyncContext';
import RecordContext from './RecordContext';
import RecordsSelectorContext from './RecordsSelectorContext';
import TabbedContext from './TabbedContext';
import ScrollToSectionContext from './ScrollToSectionContext';

export const useAccountContext = () => useContext(AccountContext);
export const useFormErrorsContext = () => useContext(FormErrorsContext);
export const useFormValuesContext = () => useContext(FormValuesContext);
export const useLocallyStoredReducerContext = () => useContext(LocallyStoredReducerContext);
export const useOrcidSyncContext = () => useContext(OrcidSyncContext);
export const useRecordContext = () => useContext(RecordContext);
export const useRecordsSelectorContext = () => useContext(RecordsSelectorContext);
export const useTabbedContext = () => useContext(TabbedContext);
export const useScrollToSectionContext = () => useContext(ScrollToSectionContext);

export {
    AccountContext,
    FormErrorsContext,
    FormValuesContext,
    LocallyStoredReducerContext,
    OrcidSyncContext,
    RecordContext,
    RecordsSelectorContext,
    ScrollToSectionContext,
    TabbedContext,
};
