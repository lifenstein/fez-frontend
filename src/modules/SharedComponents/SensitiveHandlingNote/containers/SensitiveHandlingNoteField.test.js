import SensitiveHandlingNoteField from './SensitiveHandlingNoteField';

jest.mock('../../../../context');
import { useFormValuesContext } from 'context';

function setup(testProps = {}, args = { isShallow: true }) {
    const props = {
        ...testProps,
    };

    return getElement(SensitiveHandlingNoteField, props, args);
}

describe('SensitiveHandlingNoteField', () => {
    beforeEach(() => {
        useFormValuesContext.mockImplementation(() => ({
            formValues: {},
        }));
    });

    afterEach(() => {
        useFormValuesContext.mockReset();
    });

    it('should render default view', () => {
        const wrapper = setup({});
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render disabled view', () => {
        const wrapper = setup({ disabled: true });
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
