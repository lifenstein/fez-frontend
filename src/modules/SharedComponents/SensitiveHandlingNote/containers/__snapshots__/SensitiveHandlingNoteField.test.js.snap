// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SensitiveHandlingNoteField should render default view 1`] = `
<Fragment>
  <Field
    ariaLabel="Please select sensitive handling note"
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "compare": null,
        "type": [Function],
      }
    }
    genericSelectFieldId="rek-sensitive-handling-note-id"
    itemsList={
      Array [
        Object {
          "text": "None",
          "value": "0",
        },
        Object {
          "text": "Indigenous/First Nations people should be aware that this output contains images, voices and/or names of deceased persons.",
          "value": 456855,
        },
        Object {
          "text": "Indigenous/First Nations people should be aware that this output is about women’s business.",
          "value": 456856,
        },
        Object {
          "text": "Indigenous/First Nations people should be aware that this output is about men’s business.",
          "value": 456857,
        },
        Object {
          "text": "Assessors should be aware that this output contains content related to any of the following: violence, family or domestic violence, self-harm, sexual assault, suicide, family child removal, refugee experiences, war survivor experiences or other traumatic experiences that may be distressing or harmful to some people.",
          "value": 456858,
        },
        Object {
          "text": "Assessors should be aware that this output contains content with explicit language, hate speech, nudity or sexuality, or drug use which may be confronting and potentially distressing to some people.",
          "value": 456859,
        },
        Object {
          "text": "Other",
          "value": 456860,
        },
      ]
    }
    label="Sensitive Handling Note"
    name="filesSection.sensitiveHandlingNote.id"
    onChange={[Function]}
    selectPrompt="Select a sensitive handling note"
    textFieldId="rek-sensitive-handling-note-id"
  />
</Fragment>
`;

exports[`SensitiveHandlingNoteField should render disabled view 1`] = `
<Fragment>
  <Field
    ariaLabel="Please select sensitive handling note"
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "compare": null,
        "type": [Function],
      }
    }
    disabled={true}
    genericSelectFieldId="rek-sensitive-handling-note-id"
    itemsList={
      Array [
        Object {
          "text": "None",
          "value": "0",
        },
        Object {
          "text": "Indigenous/First Nations people should be aware that this output contains images, voices and/or names of deceased persons.",
          "value": 456855,
        },
        Object {
          "text": "Indigenous/First Nations people should be aware that this output is about women’s business.",
          "value": 456856,
        },
        Object {
          "text": "Indigenous/First Nations people should be aware that this output is about men’s business.",
          "value": 456857,
        },
        Object {
          "text": "Assessors should be aware that this output contains content related to any of the following: violence, family or domestic violence, self-harm, sexual assault, suicide, family child removal, refugee experiences, war survivor experiences or other traumatic experiences that may be distressing or harmful to some people.",
          "value": 456858,
        },
        Object {
          "text": "Assessors should be aware that this output contains content with explicit language, hate speech, nudity or sexuality, or drug use which may be confronting and potentially distressing to some people.",
          "value": 456859,
        },
        Object {
          "text": "Other",
          "value": 456860,
        },
      ]
    }
    label="Sensitive Handling Note"
    name="filesSection.sensitiveHandlingNote.id"
    onChange={[Function]}
    selectPrompt="Select a sensitive handling note"
    textFieldId="rek-sensitive-handling-note-id"
  />
</Fragment>
`;
